miri/validation.rs

```diff
@@ -455,7 +455,10 @@ impl<'a, 'mir, 'tcx: 'mir + 'a> EvalContextExt<'tcx> for EvalContext<'a, 'mir, '
                     }
 
                     // Discriminant field for enums (where applicable).
-                    Variants::Tagged { ref discr, .. } |
+                    Variants::Tagged { ref tag, .. } => {
+                        assert_eq!(i, 0);
+                        return Ok(tag.value.to_ty(tcx))
+                    },
```

Comment:
This shouldn't be duplicated, just write `tag: ref discr` like the `NicheFilling` case does with `niche`.